integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/management/AlterTableCompactionCommand.scala

```diff
@@ -128,11 +127,18 @@ case class AlterTableCompactionCommand(
       CommonUtil.readLoadMetadataDetails(carbonLoadModel)
     }
     if (compactionType == CompactionType.SEGMENT_INDEX_COMPACTION) {
+      val mergeIndexProperty = CarbonProperties.getInstance()
+        .getProperty(CarbonCommonConstants.CARBON_MERGE_INDEX_IN_SEGMENT)
+      CarbonProperties.getInstance()
```

Comment:
Better pass the boolean to `CommonUtil.mergeIndexFiles` rather than setting to `CarbonProperties`